wasm/src/main/java/com/dylibso/chicory/wasm/types/CompType.java

```diff
@@ -0,0 +1,74 @@
+package com.dylibso.chicory.wasm.types;
+
+import java.util.Objects;
+
+public final class CompType {
+    private final ArrayType arrayType;
+    private final StructType structType;
+    private final FunctionType funcType;
+
+    private CompType(ArrayType arrayType, StructType structType, FunctionType functionType) {
```

Comment:
`comptype ::= <functype> | <structtype> | <arraytype>`

Do you want to check the exclusivity as precondition?